docs: add notes about Fastify not allowing adding more routes/plugins once `ready` is called (#4330)
fastify/fastify: docs/Guides/Serverless.md

code will be executed in your serverless platform of choice with an additional
snippet of code.

**Note**:

> Fastify will not allow adding more routes or registering plugins once `ready()` is called because [Fastify will "compile" all of your plugins and routes](https://github.com/fastify/fastify/issues/1771#issuecomment-515339364).

Reviewer: ```suggestion
Fastify will not allow adding more routes or registering plugins once `ready()` 
is called because [Fastify will freeze all of your plugins and routes](https://github.com/fastify/fastify/issues/1771#issuecomment-515339364)
into a fast and immutable data structure.
```

In some serverless environments, code outside the exported handlers could be cached. This means you
cannot add routes or register plugins inside the exported handlers because it will be added more than once
to the cached Fastify instance.

> Either declare all your routes and register all your plugins outside the handlers

Reviewer: I think this sentence is too vague: if I create a new fastify, how could I start it?

So you can decide which option is best for you, but you can test both libraries.

**Note**:

> Fastify will not allow adding more routes or registering plugins once `ready()` is called. See [here](https://github.com/fastify/fastify/issues/1771#issuecomment-515339364)

Reviewer 2: ```suggestion
Fastify will not allow adding more routes or registering plugins once `ready()` is called. See [here](https://github.com/fastify/fastify/issues/1771#issuecomment-515339364).
```

> Fastify will not allow adding more routes or registering plugins once `ready()` is called. See [here](https://github.com/fastify/fastify/issues/1771#issuecomment-515339364)

Reviewer 3: Please do not use a "click here" link. https://granicus.com/blog/why-click-here-links-are-bad/

> In some serverless environment, codes outside the exported handlers could be cached. This means you

Reviewer 2: ```suggestion
In some serverless environments, codes outside the exported handlers could be cached. This means you
```

Reviewer 4: Also "codes" should be code

```suggestion
In some serverless environments, code outside the exported handlers could be cached. This means you
```

cannot add routes or register plugins inside the exported handlers because it will be added more than once
to the cached Fastify instance.

Either declare all your routes and register all your plugins outside the handlers

> or initialize new instance of Fastify inside the handlers instead. See [here](https://github.com/fastify/fastify/issues/4328)

Reviewer 2: ```suggestion
or initialize a new instance of Fastify inside the handlers instead. See [here](https://github.com/fastify/fastify/issues/4328).
```
